### ISSUE TITLE
Different lifecycle for event triggering between Chrome and shimmed browsers

### ISSUE THREAD

**Issue by dhh** [guybedford/es-module-shims#161]
Given this page:

```html
<!DOCTYPE html>
<html>
<head>
  <script async src="https://ga.jspm.io/npm:es-module-shims@0.12.6/dist/es-module-shims.js"></script>
  <script type="importmap">
    { "imports": { "turbo": "http://esm.sh/@hotwired/turbo" } }
  </script>
  <script type="module">
    import "turbo"
    document.addEventListener("turbo:load", () => { document.body.textContent = "Turbo loaded" })
  </script>
</head>
<body>
  Turbo didn't load
</body>
</html>
```

Chrome 92 will correctly change the body text to "Turbo loaded", but the shimmed browsers (tried Firefox and Safari) will not. Meaning that something in the loading/initialization phase is different.

If I load turbo through a non-esm setup, it works in all browsers as expected with:

```html
<!DOCTYPE html>
<html>
<head>
  <script src="https://cdn.jsdelivr.net/npm/@hotwired/turbo@7.0.0-rc.1/dist/turbo.es5-umd.min.js" defer></script>
  <script>
    document.addEventListener("turbo:load", () => { document.body.textContent = "Turbo loaded" })
  </script>
</head>
<body>
  Turbo didn't load
</body>
</html>
```

**Comment by guybedford** [comment]
My first thought is that this could be a case of https://github.com/guybedford/es-module-shims/issues/99.

Does Turbo attach to DOMContentLoaded?

I've suggested a fix for this previously in https://github.com/guybedford/es-module-shims/issues/99#issuecomment-881967681.

**Comment by guybedford** [comment]
Right, so I can see in the source Turbo is attaching to `document.addEventListener('readystatechange')`.

Strictly speaking, module scripts don't block domready, for example:

```html
<!doctype html>
<script type="module">
  console.log('Ready State Before: ' + document.readyState);
  document.addEventListener('readystatechange', () => {
    console.log('Ready state change before listener');
  });
  await new Promise(resolve => setTimeout(resolve, 10000));
  console.log('Ready State After: ' + document.readyState);
  document.addEventListener('readystatechange', () => {
    console.log('Ready state change after listener');
  });
</script>
```

Will trigger the "ready state before listener" but not the "ready state after listener".

So readystate attachments in modules can in theory be racy.

The way to avoid this is not to rely on catching the event, but simultaneously checking `document.readyState` along with adding the listener if it's not yet in the desired state, as described in https://github.com/guybedford/es-module-shims/issues/99#issuecomment-881967681.

**Comment by guybedford** [comment]
I've posted https://github.com/hotwired/turbo/pull/346 if it helps.

**Comment by guybedford** [comment]
Fix released in 0.12.7.
